bcm283x/gpio.go

```diff
@@ -369,29 +370,17 @@ func (p *Pin) SetFunc(f pin.Func) error {
 // possible to 'read back' what value was specified for each pin.
 //
 // Will fail if requesting to change a pin that is set to special functionality.
-//
-// Using edge detection requires opening a gpio sysfs file handle. On Raspbian,
-// make sure the user is member of group 'gpio'. The pin will be exported at
-// /sys/class/gpio/gpio*/. Note that the pin will not be unexported at
-// shutdown.
-//
-// For edge detection, the processor samples the input at its CPU clock rate
-// and looks for '011' to rising and '100' for falling detection to avoid
-// glitches. Because gpio sysfs is used, the latency is unpredictable.
 func (p *Pin) In(pull gpio.Pull, edge gpio.Edge) error {
 	if p.usingEdge && edge == gpio.NoEdge {
-		if err := p.sysfsPin.Halt(); err != nil {
+		if err := p.ioctlPin.Halt(); err != nil {
```

Comment:
Oh I see here what the code crashed if sysfsPin was nil. Oops. I mostly worry for users on old distro.